hypothesis-python/docs/compatibility.rst

```diff
@@ -25,80 +22,71 @@ changes in patch releases.
+If you experience issues running Hypothesis on other operating systems, we are
+happy to accept bug reports which either clearly point to the problem or contain
+reproducing instructions for a Hypothesis maintainer who does not have the ability
+to run that OS. It's hard to fix something we can't reproduce!
 
 .. _framework-compatibility:
 
 Testing frameworks
 ------------------
 
-In general Hypothesis goes to quite a lot of effort to generate things that
-look like normal Python test functions that behave as closely to the originals
-as possible, so it should work sensibly out of the box with every test framework.
-
-If your testing relies on doing something other than calling a function and seeing
-if it raises an exception then it probably *won't* work out of the box. In particular
-things like tests which return generators and expect you to do something with them
-(e.g. nose's yield based tests) will not work. Use a decorator or similar to wrap the
-test to take this form, or ask the framework maintainer to support our
-:ref:`hooks for inserting such a wrapper later <custom-function-execution>`.
-
-In terms of what's actually *known* to work:
-
-* Hypothesis integrates as smoothly with pytest and unittest as we can make it,
-  and this is verified as part of the CI.
-* :pypi:`pytest` fixtures work in the usual way for tests that have been decorated
-  with |@given| - just avoid passing a strategy for
-  each argument that will be supplied by a fixture.  However, function-scoped fixtures
-  will run only once for the whole function, not per example. To proactively warn you about
-  this case, we raise |HealthCheck.function_scoped_fixture|, unless suppressed with
-  |settings.suppress_health_check|.
-* The :func:`python:unittest.mock.patch` decorator works with
-  |@given|, but we recommend using it as a context
-  manager within the decorated test to ensure that the mock is per-test-case
-  and avoid poor interactions with Pytest fixtures.
-* Nose works fine with Hypothesis, and this is tested as part of the CI. ``yield`` based
-  tests simply won't work.
-* Integration with Django's testing requires use of the :ref:`hypothesis-django` extra.
-  The issue is that in Django's tests' normal mode of execution it will reset the
-  database once per test rather than once per example, which is not what you want.
-* :pypi:`coverage` works out of the box with Hypothesis; our own test suite has
-  100% branch coverage.
+In general, Hypothesis goes to quite a lot of effort to return a function from |@given| that behaves as closely to a normal test function as possible. This means that most things should work sensibly with most testing frameworks.
+
+Maintainers of testing frameworks may be interested in our support for :ref:`custom function execution <custom-function-execution>`, which may make some Hypothesis interactions possible to support.
+
+pytest
+~~~~~~
+
+The main interaction to be aware of between Hypothesis and :pypi:`pytest` is fixtures.
```

Comment:
I think it's worth calling out at the end of this section that you can apply `@given()` and `@pytest.mark.parametrize()` to the same test, with `@given()` on the inside, and so long as the supplied parameters don't overlap it works fine.  (indeed also with fixtures!)

Reply:
apparently `@given` works on the outside as well? which I did not know until I tested:

```python
@given(st.integers())
@pytest.mark.parametrize("s", ["a"])
def test_use_parametrize(s, n):
    assert isinstance(s, str)
    assert isinstance(n, int)
```